File: programmers/C++/Lv3/숫자 게임.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> A, vector<int> B) {
    int answer = 0;
    
    int cur_A = 0;
    int cur_B = 0;
    
    sort(A.begin(), A.end(), greater<int>());
    sort(B.begin(), B.end(), greater<int>());
    
    while (true){
        
        // 종료 조건
        if (cur_A == A.size() || cur_B == B.size()){
            break;
        }
        
        // B 값이 더 크다면
        if (A[cur_A]<B[cur_B]){
            answer++;
            cur_B++;
        }
        
        // A 값은 계속 커져야 함
        cur_A++;
    }
    
    return answer;
}

File: programmers/Python/Lv3/숫자 게임.py
def solution(A, B):
    A.sort(reverse=True)
    B.sort(reverse=True)
    
    answer = 0
    point_a, point_b = 0, 0
    
    while point_a!=len(A) and point_b!=len(B):
        if B[point_b] > A[point_a]:
            answer+=1
            point_b+=1
            point_a+=1
        else:
            point_a+=1
            
    return answer